homeassistant/helpers/intent.py

```diff
@@ -252,25 +272,87 @@ def __init__(
         self.text_input = text_input
         self.context = context
         self.language = language
+        self.category = category
 
     @callback
     def create_response(self) -> IntentResponse:
         """Create a response."""
         return IntentResponse(self, language=self.language)
 
 
+class IntentResponseType(Enum):
+    """Type of the intent response."""
+
+    ACTION_DONE = "action_done"
+    """Intent caused an action to occur"""
+
+    QUERY_ANSWER = "query_answer"
+    """Response is an answer to a query"""
+
+    ERROR = "error"
+    """Response is an error"""
+
+
+class IntentResponseErrorCode(str, Enum):
+    """Reason for an intent response error."""
+
+    NO_INTENT_MATCH = "no_intent_match"
+    """Text could not be matched to an intent"""
+
+    NO_VALID_TARGETS = "no_valid_targets"
+    """Intent was matched, but no valid areas/devices/entities were targeted"""
+
+    FAILED_TO_HANDLE = "failed_to_handle"
+    """Unexpected error occurred while handling intent"""
+
+
+class IntentResponseTargetType(str, Enum):
+    """Type of target for an intent response."""
+
+    AREA = "area"
+    DEVICE = "device"
+    ENTITY = "entity"
+
+
+@dataclass
+class IntentResponseTarget:
+    """Main target of the intent response."""
+
+    name: str
+    target_type: IntentResponseTargetType
+
+
+@dataclass
+class IntentResponseError:
+    """Details of an intent response error."""
+
+    code: IntentResponseErrorCode
+    message: str
+
+
 class IntentResponse:
     """Response to an intent."""
 
     def __init__(
-        self, intent: Intent | None = None, language: str | None = None
+        self,
+        intent: Intent | None = None,
+        language: str | None = None,
     ) -> None:
         """Initialize an IntentResponse."""
         self.intent = intent
         self.speech: dict[str, dict[str, Any]] = {}
         self.reprompt: dict[str, dict[str, Any]] = {}
         self.card: dict[str, dict[str, str]] = {}
         self.language = language
+        self.success = True
```

Comment:
I wonder if we still need `success` since that can be derived from `response_type`